index.js

```diff
@@ -932,25 +944,27 @@
   //. false
   //. ```
   var equals = (function() {
-    //  $seen :: Array Any
-    var $seen = [];
+    //  $pairs :: Array (Pair Any Any)
+    var $pairs = [];
 
-    //  equal :: (a, b) -> Boolean
-    var equal = function(x, y) {
-      $seen.push(x, y);
+    return function equals(x, y) {
+      if (type(x) !== type(y)) {
```

Comment:
I :heart: the simplicity of the approach. _Are you two members of the same type? Is that type a setoid? Okay, we'll let its `fantasy-land/equals` method determine whether you are equal._